src/sys/wait.rs

```diff
@@ -215,24 +236,94 @@ fn decode(pid : Pid, status: i32) -> WaitStatus {
     }
 }
 
-pub fn waitpid<P: Into<Option<Pid>>>(pid: P, options: Option<WaitPidFlag>) -> Result<WaitStatus> {
-    use self::WaitStatus::*;
+#[derive(Debug, Copy, Clone, Eq, PartialEq, Hash)]
+/// Designates whether the supplied `Pid` value is a process ID, process group ID,
+/// specifies any child of the current process's group ID, or any child of the current process.
+pub enum PidGroup {
+    ProcessID(u32),
```

Comment:
It's kind of odd to use something other than `u32`, as the purpose of separating the variants is to eliminate the need to express meaning through negative values, as `Pid` is basically `i32`. A negative value indicates a process group ID, where as a positive value indicates a process ID. If the process ID and group ID are going to get their own variants, it wouldn't make sense to require them to retain their positive and negative values. If you retained the `Pid` value, then there'd be less meaning to having both a PID and PGID variant when they'd always get the same value and that value would be treated the same by functions.

Especially when I would, for example, create a `PidGroup` manually in Ion by just supplying my existing `u32(Linux/BSD)/usize(Redox)` values that I have stored in the shell to the `ProcessGroupID` variant, rather than having to do as I do now and convert the `u32` into an `i32`, and then set the negative bit to signify to the function that I'm wanting to wait/signal a process group. Being able to just designate the PGID with a positive value is more idiomatic.